### PR TITLE
Replace errno mod with io::Error and use Result<>

### PR DESCRIPTION
**Pull request by hestela**
io::Error provides last_os_error() which is equivalent to calling errno
and strerro. One annoyance of using io::Error is that to use the
description function, you have to import error::Error which means you
have to import it under another name to get the error::Error trait.

Using io::Error reduces the need for some libc functions and allows
error numbers to be stored. io::Errors can be composed of each other,
allowing for a chain of different errors to be passed up the chain.

Will now prefer to use Result over Option when errno() will be set in
error conditions.

### CODE REVIEW

File: src/fileops.rs [hunk 1]
```diff
@@ -6,27 +6,30 @@ use std::ffi::CString;
 use libc::c_int;
 use std::os::unix::io::{RawFd, AsRawFd};
 use std::ops::Drop;
+use std::io::Error;
+use std::error::Error as std_error;
```
Comment (hestela) [inline src/fileops.rs:10]: Thanks rust for having multiple Error s that i need to import.... at least i dont have to use error::Error s... yet(?)

Comment (acmcarther) [inline src/fileops.rs:10]: at least you can alias them. More than you can say for Java.

I rly enjoy my life when I've got guava.Optional and java.util.Optional, and you have to refer to one of them fully-qualified.

File: src/fileops.rs [hunk 2]
```diff
@@ -49,6 +52,8 @@ impl Drop for Fd {
 #[test]
 // Check open/close on a file that should exist in most linux based OS.
 fn open_close_devnull() -> () {
-  let fd = Fd::open("/dev/null", libc::O_RDONLY);
-  assert!(fd.is_some());
+  match Fd::open("/dev/null", libc::O_RDONLY) {
+    Err(e) => panic!(e.description().to_owned()),
```
Comment (acmcarther) [inline src/fileops.rs:56]: why panic? for message?

**Comment by acmcarther** [comment]
lgtm

File: src/fileops.rs (hunk 2 above)
Comment (hestela) [inline src/fileops.rs:56]: I think I saw panic in a test example a while back, is there a better way to say the test failed?

Comment (acmcarther) [inline src/fileops.rs:56]: I guess not. `Result#expect` is slightly more conventional, but you want the actual error so you cant do that. Guess this'll have to do.

**Pull request merged**